### PR TITLE
style: responsive webui [DET-4417, DET-4420]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Update the WebUI to be mobile friendly. Key areas of updates:

- Primary navigation
- Support for Responsive tables
- Filter groups in Section headers

Screen Shot 2020-10-28 at 11 49 00 AM
Screen Shot 2020-10-28 at 11 49 11 AM
Screen Shot 2020-10-28 at 11 50 28 AM
Screen Shot 2020-10-28 at 11 50 52 AM
Screen Shot 2020-10-28 at 11 50 52 AM
Screen Shot 2020-10-28 at 11 51 11 AM

## Test Plan

- Check every WebUI page to ensure they render properly in desktop, tablet and mobile views (check to make sure they are fluid as well)
  - Login page
  - Dashboard
  - Experiment List
  - Experiment Detail
  - Trial Detail
  - Trial Logs
  - Task List
  - Cluster
  - Master Logs
- Check for the tables to render properly for mobile views.
- Check for the filter groups to render properly. (the filter options that appear across from the Section headers. For example the `User` or `State` filters on the experiment list page)

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/hooks/useNotebookLauncher.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react';
```
Comment (hkang1) [inline webui/react/src/hooks/useNotebookLauncher.ts:1]: convert to module (nothing in here is React specific)

File: webui/react/.storybook/main.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ module.exports = {
     '@storybook/addon-links',
     '@storybook/addon-knobs',
   ],
-  stories: ['../src/**/*.stories.(ts|tsx)'],
+  stories: ['../src/**/*.stories.@(ts|tsx)'],
```
Comment (hkang1) [inline webui/react/.storybook/main.js:22]: breaking change for Storybook 6

**Review by hamidzr — COMMENTED** [review]
Great job with making the UI responsible. This should make the WebUI a lot more usable on phones and smaller windows. I added all the comments  I had. I realized you had 2 or 3 comments open as well (eg turning that hook into a module). 

- does make storybook-build passes for you? it is passing in CI but runs into an issue for me

Screen Capture_select-area_20201102144223



- no need to mention `/det` in routes (would probably need a rebase)

### style nits

- I think the avatar could be a bit bigger compared to the logo
Screen Capture_select-area_20201102151734
alignment here
Screen Capture_select-area_20201102151919
the page itself is horizontally scrollable for experiments list
Screen Capture_select-area_20201102152049

File: webui/react/config-overrides.js [hunk 3]
```diff
@@ -13,6 +14,8 @@ const path = require('path');
 const webpack = require('webpack');
 const jestConfig = require('./jest.config');
 
+const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/config-overrides.js:17]: removed

File: webui/react/src/pages/SignIn.tsx [hunk 4]
```diff
@@ -73,13 +74,16 @@ const SignIn: React.FC = () => {
    * This will prevent the form from showing for a split second when
    * accessing a page from the browser when the user is already verified.
    */
-  return auth.checked ?
-    <div className={css.base}>
-      <div className={css.content}>
-        <Logo type={LogoTypes.OnLightVertical} />
-        <DeterminedAuth />
+  return auth.checked ? (
+    <Page>
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/pages/SignIn.tsx:79]: yeah probably, some merge conflict came up here so it's resolved now

File: webui/react/src/components/NavigationTabbar.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    height: 480,
+    position: 'relative',
+    width: 320,
+  }}>
+    <div style={{
+      alignItems: 'center',
+      display: 'flex',
+      flexGrow: 1,
+      justifyContent: 'center',
+    }}>Content Area</div>
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/components/NavigationTabbar.stories.tsx:39]: using a storybook add-on to introduce responsive view into our stories

**Comment by hkang1** [comment]
> Great job with making the UI responsible. This should make the WebUI a lot more usable on phones and smaller windows. I added all the comments I had. I realized you had 2 or 3 comments open as well (eg turning that hook into a module).
> 
> * does make storybook-build passes for you? it is passing in CI but runs into an issue for me
> 
> Screen Capture_select-area_20201102144223
> 
This worked for me when I tried it just now. Based on the error, it looks like the config-override of bringing global SASS of `@import global` was not coming into play for storybook? Try again?

> * no need to mention `/det` in routes (would probably need a rebase)
> 
Yep updated them now that your changes are in. Thanks for calling it out!

> ### style nits
> * I think the avatar could be a bit bigger compared to the logo
>
>   Screen Capture_select-area_20201102151734
>
>   alignment here
>   Screen Capture_select-area_20201102151919
>
I will do another round of style refresh as another task for these style corrections, just because there are a few other places that should get addressed (Page, Page Header, Section Options, etc).


>   the page itself is horizontally scrollable for experiments list
>   Screen Capture_select-area_20201102152049

Ah good find, the TableBatch was the culprit! Fixed.

**Review by hamidzr — APPROVED** [review]
Nice work. LGTM minus one blocking comment about `external` flag.

- suggestion: pass `Link` props up as part of actions in `ActionSheet`

File: webui/react/src/hooks/useResize.ts [hunk 6]
```diff
@@ -16,8 +16,8 @@ const defaultResizeInfo = {
 
 export const DEFAULT_RESIZE_THROTTLE_TIME = 500;
 
-const useResize = (ref: RefObject<HTMLElement>): ResizeInfo => {
-  const element = ref.current;
+const useResize = (ref?: RefObject<HTMLElement> | undefined): ResizeInfo => {
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/hooks/useResize.ts:19]: lol, yep 👍

File: webui/react/src/components/NavigationTabbar.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            onClick: () => handleLaunchNotebook(),
+          },
+          {
+            icon: 'notebook',
+            label: 'Launch CPU-only Notebook',
+            onClick: () => handleLaunchNotebook(true),
+          },
+          { icon: 'logs', label: 'Master Logs', path: '/logs', popout: true },
+          { icon: 'docs', label: 'Docs', path: '/docs', popout: true },
+          { icon: 'cloud', label: 'API (Beta)', path: '/docs/rest-api/', popout: true },
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/components/NavigationTabbar.tsx:79]: 👍

File: webui/react/src/components/IconFilterButtons.tsx [hunk 8]
```diff
@@ -25,21 +25,25 @@ const FilterButtons: React.FC<Props> = ({ buttons, onClick }: Props) => {
...
-              className={buttonClasses.join(' ')}
-              tabIndex={0}
-              onClick={handleClick(button.id)}>
-              <Icon name={button.icon} />
-            </button>
-          </Tooltip>
-        );
-      })}
+      <div>
+        <div className={css.buttons}>
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/components/IconFilterButtons.tsx:29]: good catch, extra useless div in there 🤷

File: webui/react/src/components/ResponsiveTable.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import { Table } from 'antd';
+import { TableProps } from 'antd/es/table';
+import React, { useEffect, useRef, useState } from 'react';
+
+import useResize from 'hooks/useResize';
+
+/* eslint-disable-next-line @typescript-eslint/ban-types */
+type ResponsiveTable = <T extends object>(props: TableProps<T>) => JSX.Element;
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/components/ResponsiveTable.tsx:8]: When used in a situation like:
```
ResponsiveTable<TrialItem>
```
`TrialItem` doesn't conform to `Record<string, unknown>`

File: webui/react/src/utils/task.ts [hunk 10]
```diff
@@ -1,9 +1,32 @@
...
+import { createNotebook } from 'services/api';
 import {
   ALL_VALUE, AnyTask, CommandState, CommandTask, CommandType,
   ExperimentItem, ExperimentOld, ExperimentTask, RecentCommandTask, RecentEvent,
   RecentExperimentTask, RecentTask, RunState, Task, TaskFilters, TaskType, User,
 } from 'types';
 import { terminalCommandStates } from 'utils/types';
+import { commandToTask } from 'utils/types';
+
+export const launchNotebook = async (slots: number): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/utils/task.ts:12]: I'll put it in `/services/utils`.

[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/utils/task.ts:12]: actually, discovered that this is a high level function which touches a lot of different areas. Moving it into services (no matter what file) creates a circular dependency.

**Pull request merged**